src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -34,7 +34,8 @@ trait WorkflowManager {
    * Generates a workflow with an ID, inserts it into the store and returns the pair.
    */
   def submitWorkflow(wdl: WdlSource, inputs: WorkflowInputs): Try[ManagedWorkflow] = {
-    val managedWorkflow = generateWorkflow(wdl, inputs) map {x => ManagedWorkflow(UUID.randomUUID(), x)}
+    val workflowId = UUID.randomUUID()
+    val managedWorkflow = generateWorkflow(workflowId, wdl, inputs) map {x => ManagedWorkflow(workflowId, x)}
     // If we successfully generated a workflow and successfully inserted it into the store, return the workflow/id combo
     for (wf <- managedWorkflow; m <- workflowStore.insert(wf.id, wf.workflow)) yield wf
```

Comment:
generateWorkflow returns a Try, so I think it is either on separate lines or retaining the map. The former seems better to me since we're using forcomp magic anyways